monai/apps/nuclick/transforms.py

```diff
@@ -0,0 +1,469 @@
+    """
+
+    def __init__(self, keys: KeysCollection):
+        super().__init__(keys)
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.keys:
+            _, labels, _, _ = cv2.connectedComponentsWithStats(d[key], 4, cv2.CV_32S)
+            d[key] = labels.astype(np.uint8)
+        return d
+
+
+class ExtractPatchd(MapTransform):
+    """
+    Extracts a patch from the given image and label, however it is based on the centroid location.
+    The centroid location is a 2D coordinate (H, W). The extracted patch is extracted around the centroid,
+    if the centroid is towards the edge, the centroid will not be the center of the image as the patch will be
+    extracted from the edges onwards
+
+    Args:
+        keys: image, label
+        centroid_key: key where the centroid values are stored
+        patch_size: size of the extracted patch
+    """
+
+    def __init__(self, keys: KeysCollection, centroid_key: str = "centroid", patch_size: int = 128):
+        super().__init__(keys)
+        self.centroid_key = centroid_key
+        self.patch_size = patch_size
+
+    def __call__(self, data):
+        d = dict(data)
+
+        centroid = d[self.centroid_key]  # create mask based on centroid (select nuclei based on centroid)
+        roi_size = (self.patch_size, self.patch_size)
+
+        for key in self.keys:
+            img = d[key]
+            x_start, x_end, y_start, y_end = self.bbox(self.patch_size, centroid, img.shape[-2:])
+            cropped = img[:, x_start:x_end, y_start:y_end]
+            d[key] = self.pad_to_shape(cropped, roi_size)
+        return d
+
+    def bbox(self, patch_size, centroid, size):
+        x, y = centroid
+        m, n = size
+
+        x_start = int(max(x - patch_size / 2, 0))
+        y_start = int(max(y - patch_size / 2, 0))
+        x_end = x_start + patch_size
+        y_end = y_start + patch_size
+        if x_end > m:
+            x_end = m
+            x_start = m - patch_size
+        if y_end > n:
+            y_end = n
+            y_start = n - patch_size
+        return x_start, x_end, y_start, y_end
+
+    def pad_to_shape(self, img, shape):
+        img_shape = img.shape[-2:]
+        s_diff = np.array(shape) - np.array(img_shape)
+        diff = [(0, 0), (0, s_diff[0]), (0, s_diff[1])]
+        return np.pad(img, diff, mode="constant", constant_values=0)
+
+
+class SplitLabeld(Transform):
+    """
+    Extracts a single label from all the given classes, the single label is defined by mask_value, the remaining
+    labels are kept in others
+
+    Args:
+        label: label source
+        others: other labels storage key
+        mask_value: the mask_value that will be kept for binarization of the label
+        min_area: The smallest allowable object size.
+    """
+
+    def __init__(self, label: str = "label", others: str = "others", mask_value: str = "mask_value", min_area: int = 5):
```

Comment:
I would suggest storing constants, like the default string values here, in an `Enum` somewhere like we do in `monai.utils.enums` so that it's clear to users what the possible values are.

Reply:
This has not been for any of the transforms that are app-based for e.g Deepgrow or Deepedit.  Also, these transforms are just meant to be used directly with the NuClick app in MONAI Label.